azurerm/internal/services/sentinel/sentinel_alert_rule_data_source.go

```diff
@@ -51,17 +53,22 @@ func dataSourceSentinelAlertRuleRead(d *schema.ResourceData, meta interface{}) e
 	resp, err := client.Get(ctx, workspaceID.ResourceGroup, "Microsoft.OperationalInsights", workspaceID.WorkspaceName, name)
 	if err != nil {
 		if utils.ResponseWasNotFound(resp.Response) {
-			return fmt.Errorf("Sentinel Alert Rule %q (Resource Group %q / Workspace: %q) was not found", name, workspaceID.ResourceGroup, workspaceID.WorkspaceName)
+			return fmt.Errorf("Sentinel Alert Rule %q (Workspace: %q) was not found", name, workspaceID)
 		}
 
-		return fmt.Errorf("retrieving Sentinel Alert Rule %q (Resource Group %q / Workspace: %q): %+v", name, workspaceID.ResourceGroup, workspaceID.WorkspaceName, err)
+		return fmt.Errorf("retrieving Sentinel Alert Rule %q (Workspace: %q): %+v", name, workspaceID, err)
 	}
 
 	id := alertRuleID(resp.Value)
```

Comment:
since we're managing our own ID and are only concerned that this exists within the API - we don't need to rely on the one returned from the API here - so this can become:

```suggestion
	id :=  parse.NewAlertRuleID(subscriptionId, resourceGroup, workspaceName, name)
	d.SetID(id.ID())
```